File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        sorted_str = sorted(strs)
        first = sorted_str[0]
        last = sorted_str[-1]
        res = ""
        for i in range(len(first)):
            if first[i] != last[i]:
                return res
            res += first[i]
        return res
        # min_str = min(strs, key = len)
        # if not strs:
        #     return ""
        
        # for idx, char in enumerate(min_str):
        #     for word in strs:
        #         if word[idx] != char:
        #             return min_str[:idx]
        # return min_str

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        String first = strs[0];
        String last = strs[strs.length-1];
        int cnt = 0;
        for(int i = 0; i < Math.min(first.length(), last.length()); i++){
            if(first.charAt(i) == last.charAt(i)){
                cnt++;
            }else{
                break;
            }
        }
        return first.substring(0, cnt);

    //     if(strs == null || strs.length == 0){
    //         return "";
    //     }
    //     Arrays.sort(strs);
    //     String first = strs[0];
    //     String last = strs[strs.length-1];
    //     StringBuilder res = new StringBuilder();
    //     for(int i =0; i < first.length();i++){
    //         if(first.charAt(i) != last.charAt(i)){
    //             return res.toString();
    //         }res.append(first.charAt(i));
    //     }
    //     return res.toString();
    }
}